app/controllers/providers/proceedings_types_controller.rb

```diff
@@ -23,8 +23,11 @@ def create
     # PATCH /provider/applications/:legal_aid_application_id/proceedings_types/:id
     def update
       authorize legal_aid_application
-      legal_aid_application.proceeding_types.clear # Remove this when multiple proceeding types required!
-      legal_aid_application.proceeding_types << proceeding_type unless legal_aid_application.proceeding_types.include?(proceeding_type)
+      ActiveRecord::Base.transaction do
+        legal_aid_application.reset_proceeding_types! # This will probably change when multiple proceeding types implemented!
+        legal_aid_application.proceeding_types << proceeding_type unless legal_aid_application.proceeding_types.include?(proceeding_type)
```

Comment:
I left it in because we are seeing very weird behaviour whereby two or more PATCH requests are sometimes being sent (if the number of threads on puma is greater than 1 - but we've also seen it on UAT.) I presumed that you left it there on purpose.  I've created a ticket for us to investigate this issue a bit further - AP-826.  Any insight you can provide is very welcome!

Reply:
Julien cleverly found out what was causing this (JS nonsense as I suspected), so I'll take this out now.